lib/pbench/server/api/resources/__init__.py

```diff
@@ -991,7 +1004,11 @@ def _dispatch(
             except UnauthorizedAccess as e:
                 self.logger.warning("{}", e)
                 abort(e.http_status, message=str(e))
-        return method(new_data, request)
+        try:
+            response_data = method(new_data, request)
+            return response_data
+        except APIAbort as e:
+            abort(e.http_status, message=str(e.message))
```

Comment:
Whenever the `APIAbort` exception is raised, there is already `self.logger.exception(..)` triggered. In that case, It'll be a repetitive one. If needed we can add `self.logger.exception(..)` in the `APIAbort` as well .

Reply:
That's not really true, as I've pointed out in several other examples.

Reply:
Per the [Flask docs](https://flask.palletsprojects.com/en/2.1.x/errorhandling/#handling):
> When Flask catches an exception while handling a request, it is first looked up by code. If no handler is registered for the code, Flask looks up the error by its class hierarchy; the most specific handler is chosen. If no handler is registered, [HTTPException](https://werkzeug.palletsprojects.com/en/2.1.x/exceptions/#werkzeug.exceptions.HTTPException) subclasses show a generic message about their code, while other exceptions are converted to a generic “500 Internal Server Error”.

So, other than for logging and clean-up requirements, we don't need to handle "internal errors" at all.

Reply:
Yeah, looks like it uses a `StreamHandler` to `stderr` by default, with `%s` formatting; so we'd need to do some default configuration work (e.g., with `dictConfig` in the documentation page you linked) in order to get Flask's logging into shape. All of that is way too much for this PR.

Going back to my initial comment here, we don't need the explicit `str` conversion on the `e.message`, and a `self.logger.exception` on an `Exception` that gets to here would probably be handy rather than completely losing the cause of the unexpected failure.